### PR TITLE
Feature/scroll watch aside

### PR DESCRIPTION
**Pull request by fraincs**
- Merged PR #28 
- Added scroll-aware marker on Aside
- Fix for Main overflowing when the main element had too wide code samples. The current fix prevent heading links visibility.

### CODE REVIEW

File: apps/docs/components/Card/Card.tsx
Comment (franckgaudin) [inline apps/docs/components/Card/Card.tsx:1]: Rename de folder `Card` by `card` and move to the `ui` folder

File: apps/docs/components/Card/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { ReactNode } from "react";
+import "./card.css";
+
+export function Card({ children }: { children: ReactNode }) {
```
Comment (franckgaudin) [inline apps/docs/components/Card/Card.tsx:4]: For the component use a arrow function and default export 
```suggestion
const Card = ({ children }: { children: ReactNode }) => (
```

File: apps/docs/components/Image/Image.tsx
Comment (franckgaudin) [inline apps/docs/components/Image/Image.tsx:1]: replace `Image` folder by `image` and move to the `ui` folder

File: apps/docs/components/Image/Image.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import "./image.css";
+
+interface NextImageProps extends Omit<ImageProps, "src" | "alt" | "width" | "height"> {
+    alt: string;
+    src: string;
+    width: number;
+    height: number;
+}
+
+export const NextImage: React.FC<NextImageProps> = ({ alt, src, width, height, ...props }) => {
```
Comment (franckgaudin) [inline apps/docs/components/Image/Image.tsx:12]: Same comment as for the Card component
```suggestion
const NextImage = ({ alt, src, width, height, ...props }: React.PropsWithoutRef<NextImageProps>) => {
```

File: apps/docs/components/ui/aside/aside.css [hunk 3]
```diff
@@ -30,19 +46,25 @@
 }
 
 .hd-aside__item {
-    padding-block: var(--hd-space-05);
     font-size: 0.875rem;
+    line-height: 1.7142857143;
```
Comment (franckgaudin) [inline apps/docs/components/ui/aside/aside.css:50]: it's a very precise value why so many digits after the decimal point?

File: apps/docs/components/ui/title/title.css [hunk 4]
```diff
@@ -2,7 +2,7 @@
   --hd-title-font-family: var(--hd-default-font-family);
 
   font-family: var(--hd-title-font-family);
-  line-height: 1.35;
+  line-height: 1;
```
Comment (franckgaudin) [inline apps/docs/components/ui/title/title.css:5]: Why did you change this value to 1?

File: apps/docs/content/pages/intro.mdx
Comment (franckgaudin) [inline apps/docs/content/pages/intro.mdx:1]: What is this page for and where is it displayed?

File: apps/docs/content/tokens/getting-started/introduction.mdx
Comment (franckgaudin) [inline apps/docs/content/tokens/getting-started/introduction.mdx:1]: The image does not work when rendered on Netlify. 

Capture d’écran, le 2023-07-26 à 20 45 41

File: apps/docs/content/tokens/getting-started/introduction.mdx [hunk 5]
```diff
@@ -8,10 +8,41 @@ order: 1
...
+
 ## Definitions
 
 **Core tokens** are referring to raw values such as an hexadecimal color code, a border-width, a font-size. Think of them as paint name, Chocolate Frost is easier to use in a conversation than the exact color recipe.
 
 **Semantic tokens** adds the intention layer to tokens, they map to a core token, or tokens should we say as the semantic tier has a notion of themability. Using our Chocolate Frost paint color example this would dictate where the paint color should be used, on dry walls or plastic, inside or outside.
 
 **Component tokens** refers to the nuts and bolts of the components and their implementation details.  They are tied to their component. In our paint color example, this would be where a specific color is actually used, in which room and on which architectural part of it(ceiling, doors, wall).
+
+```json title="package.json"
```
Comment (franckgaudin) [inline apps/docs/content/tokens/getting-started/introduction.mdx:23]: The title style seems broken here

Capture d’écran, le 2023-07-26 à 20 46 29

File: apps/docs/components/ui/aside/Aside.tsx
Comment (franckgaudin) [inline apps/docs/components/ui/aside/Aside.tsx:1]: I'd like to test this feature, is there a page where I can see it in operation?

File: apps/docs/app/globals.css [hunk 6]
```diff
@@ -28,8 +33,11 @@ a {
 }
 
 .hd-wrapper {
-    width: min(100% - 3rem, 75rem);
```
Comment (franckgaudin) [inline apps/docs/app/globals.css:31]: Why have you changed your approach here?

File: apps/docs/components/Card/Card.tsx
Comment (fraincs) [reply apps/docs/components/Card/Card.tsx:1]: Looks like a merge that went bad.

File: apps/docs/components/Image/Image.tsx
Comment (fraincs) [reply apps/docs/components/Image/Image.tsx:1]: The merge kept both will remove this one.

File: apps/docs/app/globals.css (hunk 6 above)
Comment (fraincs) [reply apps/docs/app/globals.css:31]: While debugging an issue I wrongfully thought it was part of the issue, upon further inspection I can roll back to this approach.

File: apps/docs/content/tokens/getting-started/introduction.mdx
Comment (franckgaudin) [reply apps/docs/content/tokens/getting-started/introduction.mdx:1]: do we have any idea when the CDN will be in place?

**Comment by franckgaudin** [comment]
Scrolling no longer works when you click on the links on the right. However, it works fine when you click on the title. 

https://github.com/gsoft-inc/wl-hopper/assets/2379339/4c60224b-67a8-4642-8e7c-7ad1ec8f6450

**Comment by franckgaudin** [comment]
The colors that should be below the header do not appear in safari.

safari-hopper-decorative

**Comment by fraincs** [comment]
> The colors that should be below the header do not appear in safari.
> 
> safari-hopper-decorative

good catch, `transform: translate3d(0, 0, 0);` fixes the issue.

File: apps/docs/components/ui/aside/Aside.tsx [hunk 7]
```diff
@@ -16,18 +16,102 @@ interface AsideProps {
...
+        return () => {
+            window.removeEventListener("scroll", handleScroll);
+        };
+    });
+
     const listItems = links.map(link => (
-        <li className="hd-aside__item" key={link.id}>
-            <Link href={link.url} className="hd-aside__link">
+        <li className={`hd-aside__item ${ activeSection === link.id ? "hd-aside__item--active" : "" }`} key={link.id}>
+            <a href={link.url} className="hd-aside__link" >
```
Comment (fraincs) [inline apps/docs/components/ui/aside/Aside.tsx:98]: https://github.com/vercel/next.js/issues/44139 prevents us to use <Link>

**Review by franckgaudin — APPROVED** [review]
